app/javascript/src/components/Dashboard/index.jsx

```diff
@@ -1,24 +1,47 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { Route, Redirect, Switch } from "react-router-dom";
 
+import { DUMMY_CONTACTS, DUMMY_NOTES, TAGS, ROLES } from "common/dummy";
 import Sidebar from "components/Common/Sidebar";
 import {
   DASHBOARD_ROUTES,
   NOTES_PATH,
   DASHBOARD_PATH,
 } from "components/routeConstants";
 
-const Dashboard = () => (
-  <div className="flex h-screen w-full">
-    <Sidebar />
-    <Switch>
-      {DASHBOARD_ROUTES.map(({ path, component }) => (
-        <Route exact component={component} key={path} path={path} />
-      ))}
-      <Redirect from={DASHBOARD_PATH} to={NOTES_PATH} />
-    </Switch>
-  </div>
-);
+const DashboardContext = React.createContext("dashboard");
 
-export default Dashboard;
+const Dashboard = () => {
+  const [notes, setNotes] = useState(DUMMY_NOTES);
+  const [contacts, setContacts] = useState(DUMMY_CONTACTS);
+  const [tags, setTags] = useState(TAGS);
+  const [roles, setRoles] = useState(ROLES);
+
+  return (
+    <div className="flex h-screen w-full">
+      <Sidebar />
+      <Switch>
+        <DashboardContext.Provider
+          value={{
+            notes,
+            contacts,
+            tags,
+            roles,
+            setNotes,
+            setContacts,
+            setTags,
+            setRoles,
+          }}
```

Comment:
From my understanding `notes` are only needed in the `Note` related components. `contacts` are only needed in the `Contact` related components.

If that's the case, then this context is a big overkill and very unnecessary. 

- https://blog.jannikwempe.com/when-to-not-use-react-context-api-for-state
- https://blog.logrocket.com/pitfalls-of-overusing-react-context/

Reply:
Ok, I will remove the Context API and limit the scope to respective components.